### PR TITLE
ENH: `stats.boxcox_llf`: add array API support

### PR DESCRIPTION
**Pull request by lucascolley**
#### Reference issue
Towards gh-20544

#### Additional information
Seems quite simple!

### CODE REVIEW

File: scipy/stats/_morestats.py [hunk 1]
```diff
@@ -942,25 +942,34 @@ def boxcox_llf(lmb, data):
...
-        logvar = np.log(np.var(logdata, axis=0))
+        logvar = xp.log(xp.var(logdata, axis=0))
     else:
         # Transform without the constant offset 1/lmb.  The offset does
         # not affect the variance, and the subtraction of the offset can
         # lead to loss of precision.
         # Division by lmb can be factored out to enhance numerical stability.
         logx = lmb * logdata
+        # convert to `np` for `special.logsumexp`
+        logx = np.asarray(logx)
```
Comment (mdhaber) [inline scipy/stats/_morestats.py:966]: I suppose we can merge this but I'll add to the list that we need to come back after `special.logsumexp` is done.

File: scipy/stats/_morestats.py [hunk 2]
```diff
@@ -942,25 +942,34 @@ def boxcox_llf(lmb, data):
...
-    logdata = np.log(data)
+    if xp.isdtype(data.dtype, 'integral'):
+        data = xp.asarray(data, dtype=xp.float64)
+    
+    logdata = xp.log(data)
 
     # Compute the variance of the transformed data.
     if lmb == 0:
-        logvar = np.log(np.var(logdata, axis=0))
+        logvar = xp.log(xp.var(logdata, axis=0))
```
Comment (mdhaber) [inline scipy/stats/_morestats.py:958]: Separately, we'll want to add `axis` support. Would you be willing to do that?

File: scipy/stats/tests/test_morestats.py [hunk 3]
```diff
@@ -1919,48 +1919,52 @@ def test_brack(self):
                             -0.71215366521264145, decimal=7)
 
 
+@skip_xp_backends(cpu_only=True)
+@array_api_compatible
 class TestBoxcox_llf:
 
-    def test_basic(self):
+    def test_basic(self, xp):
```
Comment (mdhaber) [inline scipy/stats/tests/test_morestats.py:1926]: Looks like there are no tests of native dtype or `float32`. What do you think of `parametriz`ing this one over those possibilities?

File: scipy/stats/_morestats.py (hunk 2 above)
Comment (lucascolley) [reply scipy/stats/_morestats.py:958]: I can try!

File: scipy/stats/tests/test_morestats.py (hunk 3 above)
Comment (lucascolley) [reply scipy/stats/tests/test_morestats.py]: Looks like we need some tol changes

Comment (mdhaber) [reply scipy/stats/tests/test_morestats.py]: I think it's just that somewhere in the `float32` calculation the result is converted to an inaccurate `float64`. The reference value is still `float64`, so the test didn't fail due to `dtype`. So if the `dtype` is not promoted accidentally, I don't think there will be a problem.

**Comment by mdhaber** [comment]
Thanks @lucascolley. I went ahead and merged. There is some follow-up work to be done. I think we need to:
1. resurrect [`logsumexp`](gh-20935)
2. add array API support to `_log_var` and `_log_mean`
3. add `axis` support

Interested in some of those?

**Comment by lucascolley** [comment]
@mdhaber could you give me some pointers for what `axis` support should look like? I'll send a PR.

**Comment by mdhaber** [comment]
Yeah! For the most part, it just means adding `axis` as a kwarg to the function and consistently passing it to the reducing functions called within (`var`, `sum`, and `logsumexp`). Let's carry that through `_log_mean` and `_log_var`, too. When the PR is up, I can suggest how to add the `_axis_nan_policy` decorator and add the standard tests. We won't worry about array API initially. How does that sound?

**Pull request merged**